src/dvc_render/vega.py

```diff
@@ -85,13 +119,39 @@ def get_filled_template(
                 value = self.template.escape_special_characters(value)
             self.template.fill_anchor(name, value)
 
-        if as_string:
-            return json.dumps(self.template.content)
+        return self.template.content
+
+    def get_partial_filled_template(self):
+        """
+        Returns a partially filled template along with the split out anchor content
+        """
+        content = self.get_filled_template(
+            split_anchors=[
+                "color",
+                "data",
+                "plot_height",
+                "plot_width",
+                "shape",
+                "stroke_dash",
+                "title",
+                "tooltip",
+                "x_label",
+                "y_label",
+                "zoom_and_pan",
+            ],
+            strict=True,
+        )
+        return content, {"anchor_definitions": self._split_content}
 
+    def get_template(self):
+        """
+        Returns unfilled template (for Studio)
+        """
         return self.template.content
 
     def partial_html(self, **kwargs) -> str:
-        return self.get_filled_template()  # type: ignore
+        content = self.get_filled_template()
+        return json.dumps(content)
```

Comment:
Why does this dump to json now?

Reply:
You answered this question [here](https://github.com/iterative/dvc-render/pull/142/files#r1412218923). Previously, the `get_filled_template` method took the `as_string` parameters, defaulting to `True`. The majority of the callers then had to set it to false. This was the only caller that used the default so I removed that option and dumped the content to a string here.

IMO, the fact that you're asking these questions confirm that the way it was written was difficult to grok.